thread/test/test.cpp

```diff
@@ -685,7 +685,7 @@ TEST(Semaphore, heavy) {
     semaphore sem(0);
     const int thread_num = 100000;
     for (int i=1; i<=thread_num;i++) {
-        thread_create11(16 * PAGE_SIZE, semaphore_heavy, sem, i);
+        thread_create11(4 * PAGE_SIZE, semaphore_heavy, sem, i);
```

Comment:
Self-hosted ARM CI machine doesn't have plenty of memory. Test might be OOM killed.